### PR TITLE
Fix compatibility with PHP 7.4

### PR DESCRIPTION
**Pull request by nikic**
The main change here is to add the `stream_set_option()` method to the stream wrapper.

Additionally fix use of null as an array and use closure instead of create_function.

### CODE REVIEW

**Comment by nikic** [comment]
I've bumped the minimum version requirement to PHP 5.3 to allow the use of closures. PEAR itself actually requires PHP >= 5.4, so I don't think that's a problem.

**Comment by nikic** [comment]
Ping. It would be good to resolve this soon, to avoid a bad PEAR experience on 7.4 :)

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,19 @@
 language: php
-php:
-  - 5.2
-  - 5.3
-  - 5.4
-  - 5.5
-  - 5.6
-  - 7
+
+matrix:
```
Comment (ashnazg) [inline .travis.yml]: @nikic are you sure about this Travis syntax?  I'm trying to separately use these exact changes to get a clean Travis biuld before applying this PR... but this syntax gives me Ruby builds every time.

Comment (nikic) [reply .travis.yml]: Weird. This is what the build looked like in my fork: https://travis-ci.com/nikic/PHP_Archive/builds/124732148 Seems to work there.

Comment (ashnazg) [reply .travis.yml]: Yeah, that one does look good... let's go with that one as the "good PR build" 👍

**Comment by nikic** [comment]
Would it be possible to release a new PHP_Archive version and use it for a new pear-core build? Just checked that install PEAR as part of PHP still throws these warnings (if phar is disabled).

**Comment by ashnazg** [comment]
I'll see if I can get to this later today.

**Comment by ashnazg** [comment]
v0.13.0 is released now

**Pull request merged**